packages/fetch-and-compile/lib/types.ts

```diff
@@ -0,0 +1,38 @@
+import type { WorkflowCompileResult } from "@truffle/compile-common";
+import type { SourceInfo } from "@truffle/source-fetcher";
+
+export type FailureType = "fetch" | "compile";
+
+export interface SingleResult {
+  compileResult: WorkflowCompileResult;
+  sourceInfo: SourceInfo;
+}
+
+export interface FetchExternalErrors {
+  fetch: string[]; //addresses
+  compile: string[]; //addresses
+  fetchers: string[]; //fetcher names
+}
+
+export interface Recognizer {
+  getUnrecognizedAddresses(): string[];
+  getAnUnrecognizedAddress(): string | undefined;
+  addCompiledInfo(
+    compileResult: WorkflowCompileResult,
+    sourceInfo: SourceInfo,
```

Comment:
These two seem like they could be merged into a `SingleResult` (or whatever we end up calling it)

Reply:
I don't understand, what do you want to be different here?  You want to mash their fields together?

Reply:
Yeah, I want to mash the fields together so we're passing around the same `SingleResult` type we use elsewhere

Reply:
I mean this would be a new type, it's not something we use elsewhere?

Reply:
oh! I'm only talking about mashing the two fields together that compose `SingleResult`... so
```typescript
addCompiledInfo(
  fetchAndCompileResult: SingleResult,
  address: string,
  fetcherName: string
): void | Promise<void>;
```

Reply:
Oh!  You mean you just want to combine them in the inputs to this particular function!  Nevermind, I thought you meant something completely different.  Cool, I'll do this.

Reply:
(Specifically, I thought you were talking not about this function, but rather about the `SingleResult` type in general; and I thought that you were saying that instead of being `{ sourceInfo, compileResult }`, should be `{ ...sourceInfo, ...compileResult }`; that's what I meant by "mash the fields together".  A pretty bizarre suggestion that'd be... XD )

Reply:
yeah I realized that just now... that would've been bizarre :)